docs/developers/feature-versioning.md

```diff
@@ -0,0 +1,131 @@
+# Feature Versioning
+
+There are two types of features in Tekton Pipeline, API-driven features and behavioural features. Both types of features comply to the [feature gates](../../api_compatibility_policy.md#feature-gates) and the [feature graduation process](../../api_compatibility_policy.md#feature-graduation-process) specified in the [API compatibility policy](../../api_compatibility_policy.md).
```

Comment:
Is there really a point to differentiate the 2 types of feature in the context of per feature flag as eventually all the features will be using individual feature flag (so that you don't really need 2 separate subsetions below)?

I uderstand  that `enable-api-field` will still exist for a while during the migration phase, maybe that is worth mentioning the behavior during & after migration.

Reply:
Thanks for the suggestion. Yeah eventually we are going to switch to all dedicated feature flag(NOTE not all will be behind Per-feature flag, for example, some behavioural feature flag would need different values, which is not the same as Per-feature flag i.e. largerResult could take different values instead of true/false) for all new features. But the existing features, which is a combinations of behavioural and API-driven features controlled by `enable-api-fields` will persist for at least ~1 deprecation cycle.

I think it is reasonable to keep the section of behavioural features since they were once the missing pieces in the documentations. And their validations are actually different from Per-feature flags. WDYT?

Reply:
TY for the review. I agree with you that users would also like to see a section of behavioural features.

> As a user I would assume all the content in the "API-driven Features" subsection does not apply to "behaviour features", but there are quite some similarities.

Would you mind pointing out what are the similarities. Is it that both of them would use a dedicated feature flag?